### PR TITLE
NN-3073 move to a multi stage build and target buster slim

### PR DESCRIPTION
**Pull request by JamieDraperUK**

### CODE REVIEW

File: backend/index.js [hunk 1]
```diff
@@ -20,17 +20,18 @@ const setupWebSecurity = require('./setupWebSecurity')
...
 const setupReactRoutes = require('./setupReactRoutes')
 const phaseNameSetup = require('./phaseNameSetup')
-
 const pageNotFound = require('./setUpPageNotFound')
 
 const { logError } = require('./logError')
 const homepageController = require('./controllers/homepage/homepage')
 
+// eslint-disable-next-line  global-require
+const setupWebpackForDev = !config.app.disableWebpack && require('./setupWebpackForDev')
```
Comment (JamieDraperUK) [inline backend/index.js:33]: There are no dev dependencies in the docker image. This allows us to bypass any unnecessary module imports.

File: Dockerfile [hunk 2]
```diff
@@ -19,16 +37,36 @@ RUN ln -snf "/usr/share/zoneinfo/$TZ" /etc/localtime && echo "$TZ" > /etc/timezo
...
+
+COPY --from=builder --chown=appuser:appgroup \
+        /app/src/dateHelpers.js ./src/dateHelpers.js
+
+COPY --from=builder --chown=appuser:appgroup \
+        /app/src/BulkAppointments/RecurringAppointments.js ./src/BulkAppointments/RecurringAppointments.js
+
 
 ENV PORT=3000
+ENV DISABLE_WEBPACK=true
```
Comment (JamieDraperUK) [inline Dockerfile:68]: All dev dependencies have been removed due to running: `npm prune --production`

File: backend/index.js [hunk 3]
```diff
@@ -48,6 +50,9 @@ app.use(setupAuth({ oauthApi: apis.oauthApi, tokenVerificationApi: apis.tokenVer
 app.use(setupWebpackForDev())
```
Comment (petergphillips) [inline backend/index.js:50]: don't you need to remove this line too then?

File: backend/index.js [hunk 4]
```diff
@@ -45,9 +47,11 @@ app.use(setupRedirects())
 app.use(setupStaticContent())
 app.use(setupWebSession())
 app.use(setupAuth({ oauthApi: apis.oauthApi, tokenVerificationApi: apis.tokenVerificationApi }))
-app.use(setupWebpackForDev())
 
 app.use(currentUser({ prisonApi: apis.prisonApi, oauthApi: apis.oauthApi }))
+
+if (!config.app.disableWebpack) app.use(setupWebpackForDev())
```
Comment (petergphillips) [inline backend/index.js:53]: any reason for moving below the `currentUser` stuff - doesn't this mean that the webpack assets will get the current user in dev?

**Review by petergphillips — APPROVED** [review]
LGTM

File: backend/index.js (hunk 4 above)
Comment (JamieDraperUK) [reply backend/index.js:53]: No reason really. Technically a webpack module could (I think) make use of req.session, however this is just for dev so I'm not too worried.

**Pull request merged**